rets/client/client.py

```diff
@@ -3,26 +3,74 @@
 from rets.client.resource import Resource
 from rets.http import RetsHttpClient
 
+"""
+Example of metadata dict:
+
+metadata = [{
+    'ResourceID': 'Property',
+    'KeyField': 'Matrix_Unique_ID',
+    '_classes': [
+        {
+            'ClassName': 'Listing',
+            'HasKeyIndex': '1',
+            '_table': [
+                ... column fields
+            ],
+        }
+    ],
+    '_object_types': [
+        'ObjectType': 'LargePhoto',
+        'MIMEType': 'image/jpeg',
+    ]
+}, {
+    'ResourceID': 'Agent',
+    'KeyField': 'Matrix_Unique_ID',
+    '_classes': [
+        {
+            'ClassName': 'Listing',
+            'HasKeyIndex': '1',
+        }
+    ],
+}]
+"""
+
 
 class RetsClient:
 
-    def __init__(self, *args, http_client: RetsHttpClient = None, metadata: dict = None, **kwargs):
-        self._http = http_client or RetsHttpClient(*args, **kwargs)
-        self._http.login()
-        self._metadata = metadata or {}
+    def __init__(self,
+                 *args,
+                 http_client: RetsHttpClient = None,
+                 metadata: Sequence[dict] = (),
+                 capability_urls: dict = None,
+                 cookie_dict: dict = None,
+                 **kwargs):
+        self.http = http_client or RetsHttpClient(*args, **kwargs)
+        if not (capability_urls and cookie_dict):
```

Comment:
we have to pass in `login_url` or `RetsHttpClient` which I think make `capability_url`s unnecessary?